src/inject/gradeDist.js

```diff
@@ -5,153 +5,92 @@ function GradeDist(errorHandler) {
         {label: 'C', dataField: 'cPercent'},
         {label: 'D', dataField: 'dPercent'},
         {label: 'F', dataField: 'fPercent'},
-        {label: 'W', dataField: 'wPercent'},
         {label: 'P', dataField: 'pPercent'},
-        {label: 'NP', dataField: 'npPercent'},
-        {label: 'GPA', dataField: 'gpa'}
+        {label: 'NP', dataField: 'npPercent'}
     ], 'Grade Distribution data does not exist for this professor.', errorHandler);
 }
 
 GradeDist.prototype = Object.create(DataSection.prototype);
 GradeDist.prototype.constructor = GradeDist;
 
 GradeDist.prototype.getNewData = function (teacher, course, callback) {
-    var url = 'http://asucsd.ucsd.edu/gradeDistribution?' +
-        'GradeDistribution%5BTERM_CODE%5D=' +
-        '&GradeDistribution%5BSUBJECT_CODE%5D='+ course.subjectCode +
-        '&GradeDistribution%5BCOURSE_CODE%5D=' + course.courseCode +
-        '&GradeDistribution%5BCRSE_TITLE%5D=' +
-        '&GradeDistribution%5BINSTRUCTOR%5D=' + encodeURIComponent(teacher.nomiddle) +
-        '&GradeDistribution%5BGPA%5D=' +
-        '&GradeDistribution_page=1' +
-        '&ajax=gradedistribution-grid';
+    // link to professor's cape.ucsd.edu site in order to get their 6 digit page number 
+    // YQL Query URL: select href from html where url = 'https://cape.ucsd.edu/responses/Results.aspx?Name=mirza%2Cdiba&CourseNumber=cse30' and xpath = '//*[@id="ctl00_ContentPlaceHolder1_gvCAPEs_ctl02_hlViewReport"]'
+    var link = "https://query.yahooapis.com/v1/public/yql?q=select%20href%20from%20html%20where%20url%3D%22https%3A%2F%2Fcape.ucsd.edu%2Fresponses%2FResults.aspx%3FName%3D" 
+        + teacher.lname + "%252C%2B" + teacher.fname + "%26CourseNumber%3D" 
+        + course.subjectCode + course.courseCode + "%22%20and%20xpath%3D'%2F%2F*%5B%40id%3D%22ctl00_ContentPlaceHolder1_gvCAPEs_ctl02_hlViewReport%22%5D'&format=json&diagnostics=true&callback=";
+    var jsonObj;
+    
+    // Create an XMLHttpRequest object to obtain data from websites
+    var request = new XMLHttpRequest();  
```

Comment:
Instead of building your own XMLHttpRequest, I think we should use `this.fetchData` so this is consistent with all the other data sections. It can be used like:
```
this.fetchData(url, function(data) {
  // The response is now in data
})
```